web/components/FcPaneMap.vue

```diff
@@ -345,43 +384,35 @@ export default {
       }
     },
     clearHoveredFeature() {
-      if (this.hoveredFeature !== null) {
-        this.map.setFeatureState(this.hoveredFeature, { hover: false });
-        this.hoveredFeature = null;
-      }
+      this.hoveredFeature = null;
     },
     setHoveredFeature(feature) {
-      this.clearHoveredFeature();
-      if (feature !== null) {
-        this.map.setFeatureState(feature, { hover: true });
-        this.hoveredFeature = feature;
-      }
+      this.hoveredFeature = feature;
     },
     clearSelectedFeature() {
-      if (this.selectedFeature !== null) {
-        this.map.setFeatureState(this.selectedFeature, { selected: false });
-        this.selectedFeature = null;
-      }
+      this.selectedFeature = null;
     },
     setSelectedFeature(feature) {
-      this.clearSelectedFeature();
-      if (feature !== null) {
-        this.map.setFeatureState(feature, { selected: true });
-        this.selectedFeature = feature;
-      }
+      this.selectedFeature = feature;
     },
-    easeToLocation(location, oldLocation) {
-      if (location !== null) {
-        // zoom to location
-        const { lat, lng } = location;
-        const center = new mapboxgl.LngLat(lng, lat);
-        const zoom = Math.max(this.map.getZoom(), MapZoom.LEVEL_1.minzoom);
-        this.map.easeTo({
-          center,
-          duration: 1000,
-          zoom,
+    easeToLocations(locations, locationsPrev) {
+      if (locations.length > 0) {
+        // build bounding box on locations
+        const bounds = new mapboxgl.LngLatBounds();
+        locations.forEach(({ geom }) => {
+          const { coordinates, type } = geom;
+          if (type === 'Point') {
+            bounds.extend(coordinates);
+          } else if (type === 'LineString') {
+            coordinates.forEach((coordinatesPoint) => {
+              bounds.extend(coordinatesPoint);
+            });
+          }
         });
-      } else if (oldLocation === null) {
+        const cameraOptions = this.map.cameraForBounds(bounds, { padding: 40 });
```

Comment:
we should set `{ maxZoom: MapZoom.LEVEL_1.minzoom }` in the options here as well.  also separate camera options and add comment there too.